orc8r/cloud/go/services/analytics/analyzer.go

```diff
@@ -0,0 +1,182 @@
+	Schedule() error
+
+	// Run triggers the analyzer's cronjob to start running. This function
+	// blocks.
+	Run()
+}
+
+// PrometheusAnalyzer accesses prometheus metrics and performs
+// queries/aggregations to calculate various metrics
+type PrometheusAnalyzer struct {
+	Cron             *cron.Cron
+	Config           *Config
+	PrometheusClient query_api.PrometheusAPI
+	Calculations     []calculations.Calculation
+	Exporter         Exporter
+}
+
+func getRemoteCollectors() ([]protos.AnalyticsCollectorClient, error) {
+	services, err := registry.FindServices(orc8r.AnalyticsCollectorLabel)
+	if err != nil {
+		glog.Infof("err %v attempting to find remote analytics services", err)
+		return nil, err
+	}
+
+	var collectorClientList []protos.AnalyticsCollectorClient
+	for _, s := range services {
+		conn, err := registry.GetConnection(s)
+		if err != nil {
+			glog.Infof("error getting connection to remote %s err %v", s, err)
+			continue
+		}
+		collectorClientList = append(collectorClientList, protos.NewAnalyticsCollectorClient(conn))
+	}
+	return collectorClientList, nil
+}
+
+//GetPrometheusClient returns prometheus client
+func GetPrometheusClient() v1.API {
+	metricsConfig, err := config.GetServiceConfig(orc8r.ModuleName, metricsd.ServiceName)
+	if err != nil {
+		glog.Fatalf("Could not retrieve metricsd configuration: %s", err)
+	}
+	promClient, err := promAPI.NewClient(promAPI.Config{Address: metricsConfig.MustGetString(metricsd.PrometheusQueryAddress)})
+	// todo - for testing, will clean it up
+	//promClient, err := promAPI.NewClient(promAPI.Config{Address: "http://192.168.0.124:9090"})
+	if err != nil {
+		glog.Fatalf("Error creating prometheus client: %s", promClient)
+	}
+	return v1.NewAPI(promClient)
+}
+
+//NewPrometheusAnalyzer contructs a new prometheus analyzer instance
+func NewPrometheusAnalyzer(config *Config, prometheusClient v1.API, calculations []calculations.Calculation, exporter Exporter) Analyzer {
+	cronJob := cron.New()
+	return &PrometheusAnalyzer{
+		Config:           config,
+		Cron:             cronJob,
+		PrometheusClient: prometheusClient,
+		Calculations:     calculations,
+		Exporter:         exporter,
+	}
+}
+
+//Schedule method takes in a schedule string in cron format
+//and schedules the analyze job to be run at that schedule
+func (a *PrometheusAnalyzer) Schedule() error {
+	glog.Infof("Analytics job schedule %s", a.Config.AnalysisSchedule)
+
+	a.Cron = cron.New()
+	_, err := a.Cron.AddFunc(a.Config.AnalysisSchedule, a.LocalAnalyze)
+	if err != nil {
+		glog.Infof("error scheduling the local analytics function %v", err)
+		return err
+	}
+
+	_, err = a.Cron.AddFunc(a.Config.AnalysisSchedule, a.RemoteAnalyze)
+	if err != nil {
+		glog.Infof("error scheduling the remote analytics function %v", err)
+		return err
+	}
+	return nil
+}
+
+//LocalAnalyze method runs through all individual calculators
+//and exports the metrics if exporter is set
+func (a *PrometheusAnalyzer) LocalAnalyze() {
```

Comment:
We should be able to remove local analyze with the shift of the collect servicer to the `orchestrator` service

Reply:
Yes. We should be able to unify this.